core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -764,10 +769,13 @@ public void onUpdate(List<ResolvedServerInfoGroup> servers, Attributes config) {
         onError(Status.UNAVAILABLE.withDescription("NameResolver returned an empty list"));
         return;
       }
+      log.log(Level.FINE, "[{0}] resolved address: {1}, config={2}",
+          new Object[] {getLogId(), servers, config});
 
       try {
         balancer.handleResolvedAddresses(servers, config);
       } catch (Throwable e) {
+        log.log(Level.WARNING, "[" + getLogId() + "] Caught exception from LoadBalancer", e);
```

Comment:
So that we don't need to repeat the log in each LoadBalancer. Also we don't control 3rd-party LoadBalancers and cannot guarantee they will handle it or log it. This is for our own benefit, in that we always have a way to know what's happening.